libcodechecker/libhandlers/check.py

```diff
@@ -317,6 +437,10 @@ def __update_if_key_exists(source, target, key):
                           'add_compiler_defaults',
                           'clangsa_args_cfg_file',
                           'tidy_args_cfg_file',
+                          'name',
```

Comment:
What is the name here?

Reply:
This is the `name` command line argument. I don't really understand why exactly are these listed here.

Reply:
This list contains the argument which **don't** have any default value, neither in `check` nor in the wrapped subcommand. This list contains the variable names (not the optstrings).

The reason behind this is to keep the behaviour of `check` and the wrapped commands together: if the wrapped command does not provide a default value: `default=argparse.SUPPRESS`, meaning that if the optstring is missing, the resulting `Namespace` -- the argparse parsed argument dict -- **won't contain the key** for the argument in particular), then the wrapped command should not have a value set in the `args` dict either.

The wrapped commands are called directly on their `main(args)` method with an `args` specifically constructed from `check`'s `args` based on which argument should be forwarded where.

Reply:
So if you say `check --saargs ./a.txt` but do not specify `--tidyargs`, the wrap to analyze will emulate the behaviour of `analyze --saargs ./a.txt` without `--tidyargs`: just as how `tidy_args_cfg_file` (the `dest` for `--tidyargs`) is missing from `check`'s `args`, it won't be put into `analyze`'s `args` either.

The method which does this translation is basically a `setattr()` call, I think defined in the local scope of this wrapper method at the top.